pyinfra/facts/server.py

```diff
@@ -208,7 +208,7 @@ class Mounts(FactBase[Dict[str, MountsDict]]):
 
     @override
     def command(self):
-        if platform.system() == "FreeBSD":
+        if host.get_fact(Kernel).strip() == "FreeBSD":
```

Comment:
Unfortunately this won’t work (can’t call facts from facts). Will need to include the check condition within the command itself.

Also noting to self should make this error properly here, as shown by the failing test case.

Reply:
Yes, I had doubts about that change but I have the same result as @jgelens comments, it works. The other solution is to put a conditional check through `sh(1)` in the `command(...)` function but using pyinfra for this is better.

This is what happens when I run pyinfra (excerpt):

```console
...
--> Gathering facts...
    [control-centralita] Loaded fact server.Kernel
    [control-centralita] Loaded fact server.Kernel
    [control-centralita] Loaded fact server.Mounts
...
```

Reply:
https://github.com/pyinfra-dev/pyinfra/commit/9bc54b8c38b90a5a88e7eea8671477b44b7efd8c cleans the use of state/host up in the API. Facts should only ever (be definiton) be called with the host context set, so this is perfectly safe, same as code under `operations`.

Reply:
And https://github.com/pyinfra-dev/pyinfra/pull/1382/commits/b5196274a7b174b29125f118e73359a1066633f5 to get the tests working.